internal/pkg/util/fs/fuse/fuse_mount_linux.go

```diff
@@ -124,33 +131,95 @@ func (i *ImageMount) generateCmdArgs() ([]string, error) {
 		}
 	}()
 
+	opts, err := i.generateMountOpts()
+	if err != nil {
+		return args, err
+	}
+
+	if len(opts) > 0 {
+		args = append(args, "-o", strings.Join(opts, ","))
+	}
+
+	args = append(args, i.SourcePath)
+	args = append(args, i.mountpoint)
+
+	return args, nil
+}
+
+func (i ImageMount) generateMountOpts() ([]string, error) {
 	// TODO: Think through what makes sense for file ownership in FUSE-mounted
 	// images, vis a vis id-mappings and user-namespaces.
 	opts := []string{"uid=0", "gid=0"}
```

Comment:
I think this will result in 2 `uid=n,gid=n` strings in the squashfs flow which sets the user's uid and gid? It appears to work, but maybe uid, gid should become attributes. They'd naturally default to 0 if stored as int / uint.